multiPoolRewards-refactor

zsahmed opened the pull request:
Refactor to remove use of hardcoded values in `src/state/stake/farms.ts` and dynamically generate `multiRewardPools` from `useFarmRegistry` top level pools

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
   readonly rewardTokens: Token[]
 }
 
+export const useMultiRewardPool = (): MultiRewardPool[] => {
```
nit: useMultiRewardPools

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
+
+    await Promise.all(
+      farmSummaries.map(async (fs) => {
+        let poolContract = new Contract(fs.stakingAddress, DUAL_REWARDS_ABI, getProviderOrSigner(library) as any)
```
I would use `MoolaStakingRewards__factory` here.

I would also use `useProvider` hook instead of `getProviderOrSigner`

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
+
+        // last time the contract was updated - set isActive to false if it has been longer than 2 months
+        const periodFinish = await poolContract.periodFinish()
+        const isActive = Math.floor(Date.now() / 1000) - periodFinish > 5259492 ? false : true
```
store `5259492` as a constant variable somewhere.

Also you don't need `?`. just turn `>` into a `<`

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
+          try {
+            const externalStakingRewardAddr = await poolContract.externalStakingRewards()
+            externalStakingRwdAddresses.push(externalStakingRewardAddr)
+            poolContract = new Contract(
```
ditto from comment above

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
+            rewardsToken.push(await poolContract.rewardsToken())
+          } catch (e) {
+            //set true when externalStakingRewards() throws an error
+            baseContractFound = true
```
I would check for the specific error we expect to see. If there's an unexpected error, we should log it `console.error`

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
+    await Promise.all(
+      farmSummaries.map(async (fs) => {
+        let poolContract = new Contract(fs.stakingAddress, DUAL_REWARDS_ABI, getProviderOrSigner(library) as any)
+        const rewardsToken = []
```
nit: `rewardsTokens`

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
+          }
+        }
+
+        if (rewardsToken.length > 1) {
```
Perhaps you should check `externalStakingRewardAddresses` length instead.

BriungRi commented on src/state/stake/hooks.ts:
```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
+          multiRwdPools.push({
+            address: fs.stakingAddress,
+            underlyingPool: externalStakingRwdAddresses[0],
+            basePool: externalStakingRwdAddresses[1] ? externalStakingRwdAddresses[1] : externalStakingRwdAddresses[0],
```
Why not just use `externalStakingRewardAddresses[externalStakingRewardAddresses.length - 1]`?

BriungRi reviewed (approved):
Looks good!

I would considering deleting `farms.ts` and moving the types to the file they're directly used.

zsahmed commented:
> Looks good!
> 
> I would considering deleting `farms.ts` and moving the types to the file they're directly used.

thanks for the review, updated the PR and removed `farms.ts`, let me know if you'd like me to merge!
